src/utils/transactions.ts

```diff
@@ -0,0 +1,77 @@
+import { zeroAddress, zeroNumber } from './constants'
+import { SafeSignature } from './signatures'
+
+export interface SafeTransactionData {
+  readonly to: string
+  readonly value: string
+  readonly data: string
+  readonly operation: string
+  readonly safeTxGas: string
+  readonly baseGas: string
+  readonly gasPrice: string
+  readonly gasToken: string
+  readonly refundReceiver: string
+  readonly nonce: string
+}
+
+interface SafeTransactionDataPartial {
+  readonly to: string
+  readonly value: string
+  readonly data: string
+  readonly operation?: string
+  readonly safeTxGas?: string
+  readonly baseGas?: string
+  readonly gasPrice?: string
+  readonly gasToken?: string
+  readonly refundReceiver?: string
+  readonly nonce: string
+}
+
+export class SafeTransaction {
+  data: SafeTransactionData
+  signatures: Map<string, SafeSignature> = new Map()
+
+  constructor(data: SafeTransactionData) {
+    this.data = data
+  }
+
+  encodedSignatures(): string {
+    const signers = Array.from(this.signatures.keys()).sort()
+    const baseOffset = signers.length * 130
+    let staticParts = ''
+    let dynamicParts = ''
+    signers.forEach((signerAddress) => {
+      const signer = this.signatures.get(signerAddress)!!
+      staticParts += signer.staticPart(/*baseOffset + dynamicParts.length / 2*/)
+      dynamicParts += signer.dynamicPart()
+    })
+    return '0x' + staticParts + dynamicParts
+  }
+}
+
+export function makeSafeTransaction({
```

Comment:
why do we need a separate function? why one can't just use `new SafeTransaction`